GitCommands/ArgumentBuilderExtensions.cs

```diff
@@ -300,5 +301,68 @@ public static void Add(this ArgumentBuilder builder, [CanBeNull, ItemCanBeNull]
                 builder.Add(objectId);
             }
         }
+
+        /// <summary>
+        /// Split arguments exceeding max length into multiple batches.
+        /// Windows by default limit arguments length less than 32767 <see cref="short.MaxValue"/>.
+        /// Implementation by <see cref="System.Diagnostics.Process"/> will have file path included in command line arguments,
+        /// as well as added quotation and space characters, so we need base length to account for all these added characters
+        /// <see href="https://referencesource.microsoft.com/#system/services/monitoring/system/diagnosticts/Process.cs,1944"/>
+        /// </summary>
+        /// <param name="builder">Argument builder instance.</param>
+        /// <param name="arguments">Arguments.</param>
+        /// <param name="baseLength">Base executable file and command line length.</param>
+        /// <param name="maxLength">Command line max length. Default is 32767 - 1 on Windows.</param>
+        /// <returns>Array of batch arguments split by max length.</returns>
+        public static ArgumentString[] BuildBatchArguments(this ArgumentBuilder builder, IEnumerable<string> arguments, int baseLength, int maxLength = short.MaxValue)
```

Comment:
Doesn't this duplicate a bunch of stuff proposed in #7237? Which are you recommending?

Reply:
For #6593 this is the smallest viable amount of code. For #7237, we need to add BatchProgress callback, so if each PR would do only its purpose. I think we should focus on this before #7237.